### PR TITLE
Combine the code of dealing with 'readonly' and 'ro'.

### PR DESCRIPTION
**Pull request by SCHEN2015**
Combine the code of dealing with 'readonly' and 'ro'.
https://github.com/containers/libpod/pull/6380#discussion_r432391376

Signed-off-by: Charles Shih <schrht@gmail.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Filter before image tests is blowing up everywhere?

**Comment by mheon** [comment]
/ok-to-test

Yeah, I think master may be broken. Investigating...

**Comment by mheon** [comment]
/approve

For reference, this PR LGTM

**Comment by mheon** [comment]
```

[+0225s] ERROR: unsatisfiable constraints:
[+0225s]   man (missing):
[+0225s]     required by: world[man]
[+0225s] Error: error building at STEP "RUN apk update && apk add man": error while running runtime: exit status 1
```

This is coming out of `podman build`

File: cmd/podman/common/volumes.go [hunk 1]
```diff
@@ -209,36 +209,21 @@ func getBindMount(args []string) (spec.Mount, error) {
...
-				case "false":
-					// RW is default, so do nothing
-				default:
-					return newMount, errors.Wrapf(optionArgError, "readonly must be set to true or false, instead received %q", kv[1])
-				}
-			default:
-				return newMount, errors.Wrapf(optionArgError, "badly formatted option %q", val)
-			}
-		case "ro", "rw":
+		case "readonly", "ro", "rw":
```
Comment (TomSweeneyRedHat) [inline cmd/podman/common/volumes.go:212]: doesn't `read-only` still need to be here?

**Comment by mheon** [comment]
The dockerfile we're trying to build is just one line:

```
FROM docker.io/library/alpine:latest
```

**Comment by mheon** [comment]
Ah, nevermind, wrong test, it's a 2 liner -
```
FROM docker.io/library/alpine:latest                                                                                                                                                                                                                       
RUN apk update && apk add man
```

**Comment by mheon** [comment]
Dockerfile works fine locally, command runs fine in an Alpine image.

File: cmd/podman/common/volumes.go (hunk 1 above)
Comment (SCHEN2015) [reply cmd/podman/common/volumes.go:212]: Tom, thanks for asking.

`read-only` was in the manual, but this option had never been implemented in the code before. 

mheon replaced `read-only` with `readonly` in the manual (https://github.com/containers/libpod/pull/6380/), I went ahead and removed `read-only` from the code. As the final result, the `readonly` is in the manual and code, the same behavior as Docker.

Have a nice day.
Charles

**Comment by mheon** [comment]
Aha, found the issue. My local machine pulls Alpine 3.11, CI is pulling 3.12 - the upgrade must have broken things

**Comment by mheon** [comment]
Confirmed, manually pulling 3.12 down, I can replicate

**Comment by mheon** [comment]
@SCHEN2015 I'm going to make a PR against master to fix CI, then once that merges you'll need to rebase to pick up the fix

**Comment by SCHEN2015** [comment]
@mheon Please go ahead, I will do that. Thank you for the effort!

**Comment by mheon** [comment]
#6450 to fix CI

**Comment by mheon** [comment]
Should be good to rebase, master is green again. Thanks!

**Comment by rhatdan** [comment]
Thanks @SCHEN2015 Nice job.

**Pull request merged**